expr/http_endpoint.go

```diff
@@ -846,6 +810,72 @@ func (e *HTTPEndpointExpr) validateHeadersAndCookies() *eval.ValidationErrors {
 	return verr
 }
 
+func (e *HTTPEndpointExpr) validateSkipRequestBodyEncodeDecode() *eval.ValidationErrors {
+	verr := new(eval.ValidationErrors)
+	if !e.SkipRequestBodyEncodeDecode {
+		return verr
+	}
+	// SkipRequestBodyEncodeDecode is not compatible with gRPC or WebSocket
+	if s := Root.API.GRPC.Service(e.Service.Name()); s != nil {
+		if s.Endpoint(e.Name()) != nil {
+			verr.Add(e, "Endpoint cannot use SkipRequestBodyEncodeDecode and define a gRPC transport.")
+		}
+	}
+	if e.MethodExpr.IsPayloadStreaming() {
+		verr.Add(e, "Endpoint cannot use SkipRequestBodyEncodeDecode when method defines a StreamingPayload.")
+	}
+	if e.MethodExpr.IsResultStreaming() {
+		verr.Add(e, "Endpoint cannot use SkipRequestBodyEncodeDecode when method defines a StreamingResult. Use SkipResponseBodyEncodeDecode instead.")
+	}
+
+	// cannot define Body explicitly
+	if e.Body != nil {
+		verr.Add(e, "Cannot define a request body when using SkipRequestBodyEncodeDecode.")
+	}
+	return verr
+}
+
+func (e *HTTPEndpointExpr) validateSkipResponseBodyEncodeDecode() *eval.ValidationErrors {
+	verr := new(eval.ValidationErrors)
+	if !e.SkipResponseBodyEncodeDecode {
+		return verr
+	}
+	// SkipResponseBodyEncodeDecode is not compatible with gRPC or WebSocket.
+	if s := Root.API.GRPC.Service(e.Service.Name()); s != nil {
+		if s.Endpoint(e.Name()) != nil {
+			verr.Add(e, "Endpoint response cannot use SkipResponseBodyEncodeDecode and define a gRPC transport.")
+		}
+	}
+	if e.MethodExpr.IsPayloadStreaming() {
+		verr.Add(e, "Endpoint cannot use SkipResponseBodyEncodeDecode when method defines a StreamingPayload. Use SkipRequestBodyEncodeDecode instead.")
+	}
+	if e.MethodExpr.IsResultStreaming() {
+		verr.Add(e, "Endpoint cannot use SkipResponseBodyEncodeDecode when method defines a StreamingResult.")
+	}
+	if rt, ok := e.MethodExpr.Result.Type.(*ResultTypeExpr); ok {
+		if len(rt.Views) > 1 {
+			verr.Add(e, "Endpoint cannot use SkipResponseBodyEncodeDecode when method result type defines multiple views.")
+		}
+	}
+
+	successResp := false
+	for _, r := range e.Responses {
+		if r.Body != nil {
+			verr.Add(r, "Cannot define a response body when endpoint uses SkipResponseBodyEncodeDecode.")
+		}
+		// with SkipResponseBodyEncodeDecode at most one success response can be defined
+		if r.StatusCode < 400 {
+			if successResp && e.SkipResponseBodyEncodeDecode {
```

Comment:
This a pre-existing bug but it seems the test should also make sure that there is only a body defined (in which case there won't be a result and thus no way for Goa to know what status code to use because no field to test for the `Tag` value). If there are headers defined then I believe it's OK to have multiple responses.